### PR TITLE
Rule SLES-12-010290 - added ansible,bash and tests

### PR DESCRIPTION
**Pull request by rumch-se**
#### Description:

- _Rule SLES-12-010290 review required_

#### Rationale:

- _The rule was extended with ansible, bash and tests_

- Fixes # _Issue number here (e.g. #26) or remove this line if no issue exists._

### CODE REVIEW

File: linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_set_max_life_existing/ansible/shared.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  command: >
+    awk -F: '$5 > 60 || $5 == "" {print $1}' /etc/shadow
+  register: user_names
+  
+- name: Change the maximum time period between password changes 
+  command: >
+    passwd -x 60 {{ item }}
+  with_items: "{{ user_names.stdout_lines }}"
+  when: user_names.stdout_lines | length > 0
+
```
Comment (Mab879) [inline linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_set_max_life_existing/ansible/shared.yml:16]: Needs a new line at the end of the file.

File: linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_set_max_life_existing/bash/shared.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+# platform = multi_platform_sle
```
Comment (Mab879) [inline linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_set_max_life_existing/bash/shared.sh:3]: Please add the metadata to this file.
```
# reboot = false
# strategy = restrict
# complexity = low
# disruption = low
```

**Review by Mab879 — DISMISSED** [review]
Just a couple of minor changes needed.

Thanks for your contribution.

**Comment by rumch-se** [comment]
Hello @Mab879 and thank you for your comments. I have done the proposed changes. This commit https://github.com/ComplianceAsCode/content/pull/7368/commits/262d1bc3b08e301eb2934de8f1fe13c53a13aa84 covers them.

File: linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_set_max_life_existing/bash/shared.sh (hunk 2 above)
Comment (rumch-se) [reply linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_set_max_life_existing/bash/shared.sh:3]: Hello @Mab879 - I had done what you proposed.

File: linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_set_max_life_existing/ansible/shared.yml (hunk 1 above)
Comment (rumch-se) [reply linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_set_max_life_existing/ansible/shared.yml]: Hello @Mab879 - I had done what you proposed

Comment (Mab879) [reply linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_set_max_life_existing/ansible/shared.yml]: Looks like you have some trailing white-space on that last line. The last line of the file should be an empty (no white space).

File: linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_set_max_life_existing/bash/shared.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# strategy = restrict
+# complexity = low
+# disruption = low
+
+SECURE_MAX_PASS_AGE=60
+
+usrs_max_pass_age=( $(awk -F: '$5 > SECURE_MAX_PASS_AGE || $5 == "" {print $1}' /etc/shadow) )
+for i in ${usrs_max_pass_age[@]};
+do
+  passwd -x $SECURE_MAX_PASS_AGE $i
```
Comment (jan-cerny) [inline linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_set_max_life_existing/bash/shared.sh:14]: I see that the corresponding rule.yml says that you should run `chage` but in the remediation you run `passwd -x`. I think that the remediation and the rule.yml should be consistent, so I suggest changing either the remediation or the description in the rule.yml.

**Review by jan-cerny — APPROVED** [review]

**Review by jan-cerny — APPROVED** [review]

**Pull request merged**